src/main/java/Duke/Main/Ui.java

```diff
@@ -0,0 +1,278 @@
+                } else if (eIdx == -1 || comText.length() == (eIdx + 3)) {
+                    throw new EventTaskException();
+                } else {
+                    String desc = comText.substring(0, eIdx - 1);
+                    String at = comText.substring(eIdx + 4, comText.length()).trim();
+
+                    LocalDateTime eventDate = Parser.strToDate(at);
+                    Task event = new Event(desc, "0", eventDate);
+                    tasks.addTask(event.getStringArr());
+                    returnStr = String.format(
+                            "Got it. I've added this task:" + "\n%s\nNow you have %d tasks in the list.", event,
+                            tasks.getSize());
+                }
+                break;
+            case "find":
+                List<Task> foundTasks = tasks.findTasks(comText);
+                returnStr = String.format(
+                        "Here are the matching tasks in your list:\n%s",
+                        getList(foundTasks)
+                );
+                break;
+            case "list":
+                returnStr = getList(tasks.getList());
+                break;
+            case "done":
+                if (arr.length != 2) {
+                    throw new InvalidCommandException();
+                }
+                int doneNum = Integer.parseInt(arr[1]);
+                if (doneNum > tasks.getSize() || doneNum < 0) {
+                    throw new InvalidIndexException();
+                } else {
+                    Task item = tasks.completeTask(doneNum);
+                    returnStr = String.format("Nice! I've marked this task as done:\n%s", item);
+                }
+                break;
+            case "delete":
+                if (arr.length != 2) {
+                    throw new InvalidCommandException();
+                }
+                int deleteNum = Integer.parseInt(arr[1]);
+                if (deleteNum > tasks.getSize() || deleteNum < 0) {
+                    throw new InvalidIndexException();
+                } else {
+                    Task item = tasks.deleteTask(deleteNum);
+                    returnStr = String.format(
+                            "Noted. I've removed this task:\n%s" + "\nNow you have %d tasks in the list.", item,
+                            tasks.getSize());
+                }
+                break;
+            case "bye":
+                returnStr = "Bye. Hope to see you again soon!";
+                break;
+            default:
+                throw new InvalidCommandException();
+            }
+        } catch (Exception e) {
+            returnStr =  e.getMessage();
+        }
+        return returnStr;
+    }
+
+    /**
+     * Main command to check inputs and output the bot "answers"
+     *
+     * @param sc    Scanner to check inputs
+     * @param tasks TaskList object to manipulate database
+     */
+    public void checkCommands(Scanner sc, TaskList tasks) {
```

Comment:
Perhaps you could consider re-factoring this method and make it a bit shorter so that it is more readable.